docs/conf.py

```diff
@@ -70,6 +70,22 @@
                        'xarray': ('https://xarray.pydata.org/en/stable/', None)
                        }
 
+nitpicky = True
+nitpick_ignore = [('py:class', 'M'), ('py:class', 'N'), ('py:class', 'optional'),
+                  ('py:class', 'array_like'), ('py:class', 'array-like'),
```

Comment:
Might be good to see if there are real/better ways to declare some of these things rather than adding them to ignore. Maybe for example, use `numpy.ndarray` instead of `ndarray`, and maybe `Sequence[int]` or something like it.

Reply:
Yeah I agree that `numpy.ndarray` is probably what we should be doing. The rest of these could be better standardized too (why do we have `array-like` and `array_like`?). I'll work to try to clean up as many of these exceptions as possible.